### ISSUE TITLE
Refactor app.js to reduce its size/complexity

### ISSUE THREAD

**Issue by mossroy** [kiwix/kiwix-js#507]
The file app.js now has around 1400 lines.
It's way too much IMHO.
We should find a way to reduce its size and complexity, so that newcomers can get into it more easily. It would also be easier for us to find what we're looking for.

One approach would be to split it in different .js files.
For example, we might move some functions that are specific to one mode (jQuery or SW) in a specific file for each mode. But it's just an idea.

It might be not so easy because of some global variables that we use in app.js, that would not be accessible from other files. Maybe passing them as parameters could be enough in most cases (if the functions do not modify them)

**Comment by Jaifroid** [comment]
Good idea. It may be worth pointing out that any variables set in `init.js` become global to the whole app. It is suitable, for example, for setting global parameters retrieved from cookies, e.g. a global params object.

**Comment by ykabusalah** [comment]
Can I try refactoring this?

**Comment by Jaifroid** [comment]
@ykabusalah Unless you're familiar with the codebase, I would suggest that this is quite a complex task and not a good first issue.
